crates/shared/src/recent_block_cache.rs

```diff
@@ -217,7 +255,9 @@ where
 
         let mut cache_hit_count = 0usize;
         let mut cache_hits = Vec::new();
-        let mut cache_misses = HashSet::new();
+        // Use BTreeSet to ensure deterministic iteration order later which enables
+        // request sharing.
+        let mut cache_misses = BTreeSet::new();
```

Comment:
I think this is no longer needed because I ended up request sharing every individual request instead of a whole chunk.
Will research on next week.